packages/tools/devtools/devtools-browser-extension/src/background/BackgroundScript.ts

```diff
@@ -43,6 +43,26 @@ type Port = chrome.runtime.Port;
 
 console.log(formatBackgroundScriptMessageForLogging("Initializing Background Worker."));
 
+// Establish communication channels with the PopupScript and ContentScript.
+let contentScriptPort: Port;
+let popupPort: Port;
+
+browser.runtime.onConnect.addListener((port: Port) => {
+	if (port.name === "popup-background-channel") {
+		popupPort = port;
+		// Relay message from Popup to ContentScript
+		port.onMessage.addListener((message) => {
+			contentScriptPort?.postMessage(message);
+		});
+	} else if (port.name === "content-background-channel") {
+		contentScriptPort = port;
+		// Relay respones from ContentScript back to Popup
+		port.onMessage.addListener((message) => {
+			popupPort.postMessage(message);
```

Comment:
you have a ? in front of contentPort on line 55 but not 61. Might want to add it to 61 and/or add a warning message if there is a valid case where the port would be undefined